acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -50,3 +51,102 @@ def kill_pcp_broker(host)
     on(host, "kill -9 #{pid}")
   end
 end
+
+# Make an rpc_blocking_request to pxp-agent
+# Reference: https://github.com/puppetlabs/pcp-specifications/blob/master/pxp/request_response.md
+# @param broker the hostname or beaker host object of the machine running PCP broker
+# @param targets array of PCP identities to send request to
+#                e.g. ["pcp://client01.example.com/agent","pcp://client02.example.com/agent"]
+# @param pxp_module which PXP module to call, default pxp-module-puppet
+# @param action which action in the PXP module to call, default run
+# @param params params to send to the module. e.g for pxp-module-puppet:
+#               {:env => [], :flags => ['--noop', '--onetime', '--no-daemonize']}
+# @return hash of responses, with target identities as keys, and value being the response message for that identity as a hash
+# @raise String indicating something went wrong, test case can use to fail
+def rpc_blocking_request(broker, targets,
```

Comment:
couldn't you just use:
https://github.com/puppetlabs/ruby-pcp-client/blob/fdd6eaa9a37a0919dea1fd24c556daf5d22209a1/lib/pcp/client.rb#L44

instead of replicating a bunch of that functionality here?
i might be missing some functionality you need that isn't in that client.